src/xdsl/irdl_mlir_printer.py

```diff
@@ -0,0 +1,125 @@
+
+        elif isinstance(f, AnyTypeConstraintAttr):
+            self._print(f"Any", end="")
+
+        elif isinstance(f, DynTypeBaseConstraintAttr):
+            self._print(f.type_name.data, end='')
+
+        elif isinstance(f, DynTypeParamsConstraintAttr):
+            self._print(f.type_name.data, end='')
+            self._print("<", end="")
+            for i in range(len(f.params.data)):
+                self.print_attr_constraint(f.params.data[i])
+            self._print(">", end="")
+
+        elif isinstance(f, TypeParamsConstraintAttr):
+            self._print(f"{f.type_name}")
+
+        elif isinstance(f, NamedTypeConstraintAttr):
+            self._print(f.type_name.data, end='')
+            self._print(":", end=" ")
+            self.print_attr_constraint(f.params_constraints)
+
+        else:
+            raise Exception(f"Invalid Constraint: {type(f)} ")
+
+    def print_parameters_definition(self, param_op: ParametersOp):
+        self._print(f"      {ParametersOp.name}", end="(")
+
+        for param in param_op.constraints.data:
+            self._print(f"{param.type_name.data}: ", end="")
+            self.print_attr_constraint(param.params_constraints)
+        self._print(")")
+
+    def print_operation_definition(self, operation: OperationOp):
+        self._print(
+            f"    {OperationOp.name} {operation.attributes['name'].data} {{")
+        '''
+        Checking for existence of operands
+        '''
+        operand_list = []
+        operation.walk(lambda operand_def: operand_list.append(operand_def)
+                       if isinstance(operand_def, OperandsOp) else None)
+        if operand_list:
+            self.print_operand_definition(operand_list)
+        '''
+        Checking for existence of results
+        '''
+        result_list = []
+        operation.walk(lambda res: result_list.append(res)
+                       if isinstance(res, ResultsOp) else None)
+        if result_list:
+            self.print_result_definition(result_list)
+
+        self._print(f"    }}")
+
+    def print_result_definition(self, res_list=list[ResultsOp]):
+        self._print(f"      {ResultsOp.name}", end="(")
+
+        for i in range(len(res_list)):
+            for result in res_list[i].attributes['constraints'].data:
+                self.print_attr_constraint(result)
+                self._print(", ", end='') if i != len(res_list) - 1 else ''
+        self._print(")")
+
+    def print_operand_definition(self, op_list=list[OperandsOp]):
+        self._print(f"      {OperandsOp.name}", end="(")
+
+        for i in range(len(op_list)):
+            for ops in op_list[i].attributes['constraints'].data:
+                self.print_attr_constraint(ops)
+                self._print(", ", end='') if i != len(op_list) - 1 else ''
+        self._print(")")
+
+    def print_dialect_definition(self, di: DialectOp):
+        self._print(f"  {di.name} {di.attributes['dialect_name'].data} {{")
```

Comment:
```suggestion
        self._print(f"  {DialectOp.name} {di.attributes['dialect_name'].data} {{")
```
This is also to make things more clear, and to hopefully avoid bugs